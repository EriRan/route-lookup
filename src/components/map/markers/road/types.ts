import { Road } from "../../../../data/mapper/types";
import { ResponseSegment } from "../../../../reducers/route/calculation/types";
import { BusStopLocation } from "../../types";

export type RoadLineProps = {
  roadData: Road;
  calculationDone: boolean;
  calculatedRouteNode?: ResponseSegment;
  startPointLocation: BusStopLocation;
  endPointLocation: BusStopLocation;
  includesLines?: Array<String>;
};

export type RoadDurationProps = {
  startPointLocation?: BusStopLocation;
  endPointLocation?: BusStopLocation;
  duration: number;
};

export type RoadStyle = {
  color: string;
  opacity: number;
};
