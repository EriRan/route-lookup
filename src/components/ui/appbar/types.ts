export type KuutiolaAppBarProps = {
  classes: AppBarPropsClasses;
};

type AppBarPropsClasses = {
  centeredTitle: string;
};
