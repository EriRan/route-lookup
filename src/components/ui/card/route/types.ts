import { ActionCreatorWithPayload } from "@reduxjs/toolkit";
import { Stop } from "../../../../data/mapper/types";
import { Payload, StopState } from "../../../../reducers/route/types";

export type RouteInputProps = {
  label: string;
  onChangeFunction: ActionCreatorWithPayload<Payload, string>;
  stopMap: Map<string, Stop>;
  inputStopData: StopState | null;
  autoFocus?: boolean;
};

export type RouteInputEvent = {
  target: {
    value: unknown;
  };
};

export type CompressedRoute = {
  from: string;
  to: string;
  line: string;
};
