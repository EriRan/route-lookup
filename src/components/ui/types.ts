import { TransportData } from "../../data/mapper/types";

export type UiContainerProps = {
  transportData: TransportData;
};
