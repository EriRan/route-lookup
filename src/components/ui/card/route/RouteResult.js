import React from "react";
import { connect } from "react-redux";

import { Divider, Typography } from "@material-ui/core";

import { compressResponse } from "./routeResponseCompressor";
import {
  isUndefinedOrNull,
  isUndefinedOrNullOrEmptyString,
} from "../../../../util/Utilities";

class RouteResult extends React.Component {

  render() {
    if (!isUndefinedOrNull(this.props.calculatedRoute)) {
      return this.renderRoute(this.props.calculatedRoute);
    }
    return <div />;
  }

  renderRoute(calculatedRoute) {
    if (!isUndefinedOrNullOrEmptyString(calculatedRoute.errorMessage)) {
      return (
        <Typography key={`result-stop-no-route`}>
          {calculatedRoute.errorMessage}
        </Typography>
      );
    } else {
      const routeData = compressResponse(
        Array.from(calculatedRoute.route.values())
      ).map((stopRoute) => {
        return (
          <Typography key={`result-stop-${stopRoute.from}-${stopRoute.to}`}>
            {stopRoute.from} -> {stopRoute.to} linjalla{" "}
            {stopRoute.line}
          </Typography>
        );
      });
      return (
        <div>
          {routeData}
          <Divider />
          <Typography>
            Kesto yhteens√§: {calculatedRoute.totalDuration}
          </Typography>
        </div>
      );
    }
  }
}

const mapStateToProps = (state) => {
  return {
    calculatedRoute: state.route.calculatedRoute,
  };
};

export default connect(mapStateToProps, {})(RouteResult);
