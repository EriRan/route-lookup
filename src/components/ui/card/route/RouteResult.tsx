import { connect, ConnectedProps } from "react-redux";

import { Divider, Typography } from "@material-ui/core";

import { compressResponse } from "./routeResponseCompressor";
import { RootState } from "../../../../reducers/types";
import { CalculationResponse } from "../../../../reducers/route/change/calculation/types";
import _ from "lodash";
import RouteResultErrors from "./RouteResultErrors";
import { useTranslation } from "react-i18next";
import { CompressedRoute } from "./types";

const RouteResult = (props: Props) => {
  const { t } = useTranslation();
  if (props.calculatedRoute) {
    return renderRoute(props.calculatedRoute!);
  }
  return <div />;

  function renderRoute(calculatedRoute: CalculationResponse) {
    if (!_.isEmpty(calculatedRoute.errorMessages)) {
      return (
        <RouteResultErrors errorMessages={calculatedRoute.errorMessages} />
      );
    }
    if (!calculatedRoute.route) {
      return <div />;
    }

    const compressedRouteData = compressResponse(
      Array.from(calculatedRoute.route!.values())
    ).map((stopRoute) => {
      return (
        <Typography key={`result-stop-${stopRoute.from}-${stopRoute.to}`}>
            {renderRouteDescription(stopRoute)}
        </Typography>
      );
    });
    return (
      <div>
        {compressedRouteData}
        <Divider />
        <Typography>
          {renderTotalDuration(calculatedRoute.totalDuration)}
        </Typography>
      </div>
    );
  }

  function renderRouteDescription(stopRoute: CompressedRoute) {
    return stopRoute.from + "â†’" + stopRoute.to + " " + t("ROUTE_RESULT_WITH_LINE") + " " + stopRoute.line
  }

  function renderTotalDuration(totalDuration: number | null) {
    return t("ROUTE_RESULT_TOTAL_DURATION") + ":" + totalDuration
  }
};

const mapStateToProps = (state: RootState) => {
  return {
    calculatedRoute: state.route.calculatedRoute,
  };
};

const connector = connect(mapStateToProps, {});

type PropsFromRedux = ConnectedProps<typeof connector>;

type Props = PropsFromRedux;

export default connector(RouteResult);
