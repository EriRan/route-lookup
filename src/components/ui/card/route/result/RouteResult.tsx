import { connect, ConnectedProps } from "react-redux";

import { Divider, Typography } from "@material-ui/core";

import { compressResponse } from "./routeResponseCompressor";
import RouteResultErrorList from "./RouteResultErrorList";
import { useTranslation } from "react-i18next";
import { RootState } from "../../../../../reducers";
import { CalculationResponse } from "../../../../../reducers/route/calculation/types";
import { CompressedRoute } from "../types";
import { ROUTE_RESULT_WITH_LINE, ROUTE_RESULT_TOTAL_DURATION } from "../../../../constant/TranslationKeyConstant";

/**
 * Renders finished route calculation between two stops. If there are any errors in the calculations we render them instead
 * @param props
 * @returns
 */
const RouteResult = (props: Props) => {
  const { t } = useTranslation();
  if (props.calculatedRoute) {
    return renderRoute(props.calculatedRoute!);
  }
  return <div />;

  function renderRoute(calculatedRoute: CalculationResponse) {
    if (calculatedRoute.errorMessages.length) {
      return (
        <RouteResultErrorList errorMessageKeys={calculatedRoute.errorMessages} />
      );
    }
    if (!calculatedRoute.route.length) {
      return <div />;
    }

    const compressedRouteData = compressResponse(calculatedRoute.route).map(
      (compressedStop) => {
        return (
          <Typography key={`result-stop-${compressedStop.from}-${compressedStop.to}`}>
            {renderRouteDescription(compressedStop)}
          </Typography>
        );
      }
    );
    return (
      <div>
        {compressedRouteData}
        <Divider />
        <Typography>
          {renderTotalDuration(calculatedRoute.totalDuration)}
        </Typography>
      </div>
    );
  }

  function renderRouteDescription(stopRoute: CompressedRoute) {
    return (
      stopRoute.from +
      "â†’" +
      stopRoute.to +
      " " +
      t(ROUTE_RESULT_WITH_LINE) +
      " " +
      getLineName(stopRoute)
    );
  }

  function renderTotalDuration(totalDuration: number | null) {
    return t(ROUTE_RESULT_TOTAL_DURATION) + ":" + totalDuration;
  }

  /**
   * Return the provided line unless the line name 
   * @param line 
   * @returns 
   */
  function getLineName(stopRoute: CompressedRoute) {
    if (stopRoute.error) {
      return t(stopRoute.error);
    }
    return stopRoute.line;
  }
};

const mapStateToProps = (state: RootState) => {
  return {
    calculatedRoute: state.route.calculatedRoute,
  };
};

const connector = connect(mapStateToProps, {});

type PropsFromRedux = ConnectedProps<typeof connector>;

type Props = PropsFromRedux;

export default connector(RouteResult);
