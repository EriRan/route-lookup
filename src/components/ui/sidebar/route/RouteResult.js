import React from "react";
import { connect } from "react-redux";

import { Typography } from "@material-ui/core";
import RouteCalculator from "../../../../reducers/route/calculation/RouteCalculator";
import { compressResponse } from "./routeResponseCompressor";

import {
  isUndefinedOrNull,
  isUndefinedOrNullOrEmptyString,
} from "../../../../util/Utilities";

class RouteResult extends React.Component {
  componentDidMount() {
    this.routeCalculator = new RouteCalculator(this.props.transportData);
  }

  render() {
    if (!isUndefinedOrNull(this.props.calculatedRoute)) {
      return this.renderRoute(this.props.calculatedRoute);
    }
    return <div />;
  }

  renderRoute(calculatedRoute) {
    if (!isUndefinedOrNullOrEmptyString(calculatedRoute.errorMessage)) {
      return (
        <Typography key={`result-stop-no-route`}>
          {calculatedRoute.errorMessage}
        </Typography>
      );
    } else {
      const routeData = compressResponse(
        Array.from(calculatedRoute.route.values())
      ).map((stopRoute) => {
        return (
          <Typography key={`result-stop-${stopRoute.from}-${stopRoute.to}`}>
            Pys채kkilt채: {stopRoute.from} pys채kkiin {stopRoute.to} linjalla:{" "}
            {stopRoute.line}
          </Typography>
        );
      });
      return (
        <div>
          {routeData}
          <Typography>
            Kesto yhteens채: {calculatedRoute.totalDuration}
          </Typography>
        </div>
      );
    }
  }
}

const mapStateToProps = (state) => {
  return {
    calculatedRoute: state.route.calculatedRoute,
  };
};

export default connect(mapStateToProps, {})(RouteResult);
