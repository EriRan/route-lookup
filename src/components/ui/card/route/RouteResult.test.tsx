import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { Store, AnyAction } from 'redux';
import { createRouteLookupStore } from '../../../../reducers';
import { RootState } from '../../../../reducers/types';
import RouteResult from './RouteResult';

let store: Store<RootState, AnyAction>; 

beforeEach(() => {
  store = createRouteLookupStore();
});

test('Renders empty div when no content', () => {
  store.getState().route.calculatedRoute = null;
  // TODO: Do something about i18n translation warning here
  const component = renderer.create(
    <Provider store={store}><RouteResult /></Provider>,
  );

  expect(component).toMatchSnapshot();
});