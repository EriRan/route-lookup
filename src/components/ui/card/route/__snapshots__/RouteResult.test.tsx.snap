// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders empty div when no content 1`] = `<div />`;
